### ISSUE TITLE
[2.0.0] [PHQL] [ Bug ] Unable to set columns from joined table only

### ISSUE THREAD

**Issue by virgofx** [phalcon/cphalcon#10041]
Unable to produce a resultset that has columns only from the joined table (which should make a non-associative array of Models). 

The only way around this is to select another field from the FROM table which drops the resultset into an associative array ... however, the point is to not SELECT anything from the from table, but use it as the primary limiting query portion for the join. 

```
$resultSet = $this->getModelsManager()->createBuilder()
            ->columns('users.*')
            ->addFrom('Models\UserAccount', 'userAccount')
            ->leftJoin('Models\User','users.id = userAccount.userId', 'users')
            ->getQuery()
            ->execute();
```
`Column 'id' doesn't make part of the column map`

Adding in another column for the join as such:
```
        $resultSet = $this->getModelsManager()->createBuilder()
            ->columns('users.*, userAccount.userId')
            ->addFrom('Models\UserAccount', 'userAccount')
            ->leftJoin('Models\User','users.id = userAccount.userId', 'users')
            ->getQuery()
            ->execute();
```
Query works. The returned dataset is a `Phalcon\Mvc\Model\Resultset\Complex` with each row containing the corresponding model object for Users and a string for the key -- in this case 'userId'.

Note: any key would work from the FROM table -- it just needs 1.

The issue is that if you want to perform a join but return a row that is not nested (e.g. you simply want a toArray() to retrieve the models as values - non associative) then you currently have to have that as the FROM in the query. If the row is coming from the joined column (as in this case), PHQL requires another field from the FROM table.

Normal SQL does not require this. The equivalent syntax would be similar to the following working query. 

```
SELECT users.*
FROM `user_to_account` AS `userAccount`
LEFT JOIN `users` AS `users` ON `users`.`id` = `userAccount`.`user_id`  
```

**Comment by andresgutierrez** [comment]
Not sure if I understand correctly but the result is the expected, the following PHP returns a resultset of the specified model:

```php
<?php

namespace Some;

use Phalcon\DI,
    Phalcon\Db\Column,
    Phalcon\Mvc\Model,
    Phalcon\Events\Manager as EventsManager,
    Phalcon\Db\Adapter\Pdo\MySQL as Connection,
    Phalcon\Mvc\Model\Manager as ModelsManager,
    Phalcon\Mvc\Model\Metadata\Memory as ModelsMetaData;

$eventsManager = new EventsManager();

$di = new DI();

$connection = new Connection(array(
    "host"     => "localhost",
    "username" => "root",
    "password" => "",
    "dbname"   => "phalcon_test",
));

$connection->setEventsManager($eventsManager);

$eventsManager->attach('db',
    function ($event, $connection) {
        switch ($event->getType()) {
            case 'beforeQuery':
                echo $connection->getSqlStatement(), "\n";
                break;
        }
    }
);

$modelsManager = new ModelsManager();
$modelsManager->setDi($di);
$di['db'] = $connection;
$di['modelsManager'] = $modelsManager;
$di['modelsMetadata'] = new ModelsMetadata();

class Robots extends Model
{

}

class RobotsParts extends Model
{

}

$resultSet = $modelsManager->createBuilder()
        ->columns('rp.*')
        ->addFrom('Some\RobotsParts', 'rp')
        ->leftJoin('Some\Robots','r.id = rp.robots_id', 'r')
        ->getQuery()
        ->execute();

foreach ($resultSet as $row) {    
    echo get_class($row), PHP_EOL;
    echo $row->id, PHP_EOL;
}
```

Produces:

```sql
$ php a.php 
SELECT IF(COUNT(*)>0, 1 , 0) FROM `INFORMATION_SCHEMA`.`TABLES` WHERE `TABLE_NAME` = 'robots_parts' AND `TABLE_SCHEMA` = DATABASE()
DESCRIBE `robots_parts`
SELECT `rp`.`id`, `rp`.`robots_id`, `rp`.`parts_id` FROM `robots_parts` AS `rp` LEFT JOIN `robots` AS `r` ON `r`.`id` = `rp`.`robots_id` 
Some\RobotsParts
1
Some\RobotsParts
2
Some\RobotsParts
3
```

**Comment by virgofx** [comment]
@andresgutierrez  Almost, good example but slightly difference.  The key difference is the example works if the *columns* selection is FROM the *addFrom*. The issue is that you can not do `r.*` in your example ... which would correspond to my example.

So the problem still exists.

**Comment by andresgutierrez** [comment]
Was this working in 1.3?

**Comment by andresgutierrez** [comment]
I've added a fix in the latest commit in 2.0.0, could you please try again?

**Comment by virgofx** [comment]
@andresgutierrez I see the fix. I'm on a system that doesn't have Zephir ... any way to kick-off a rebuild of the extensions so I can rebuild and confirm. The latest CI build is just before your fix.

**Comment by andresgutierrez** [comment]
Just compile from ext/:

```php
cd cphalcon/
cd ext/
sudo ./install
```

**Comment by virgofx** [comment]
Yeah I do normally, those use the .c extensions which get generated from Travis, last ones are checked in prior to your fix :/

**Comment by virgofx** [comment]
@andresgutierrez The `print_r` bug I'm still seeing which is causing major display issues! I've recompiled from latest using zephir and same results ...

**Comment by andresgutierrez** [comment]
I wasn't able to reproduce the issue after the fix I've made

**Comment by virgofx** [comment]
Bummer @andresgutierrez ... I'm still getting the same error after updating to 16b4001c53956ba9486065af51f5596c886a5163.   Some more information.


* There do exist foreign key constraints at DB level (e.g. on the join). This shouldn't matter as the error is happening within Phalcon prior to executing the query.
* PHP - Using annotations in the models. 
* User/UserAccount same concept as Robot/RobotPart.

```sql
CREATE TABLE IF NOT EXISTS `users` (
  `id` char(36) CHARACTER SET ascii NOT NULL,
  `name` varchar(45) NOT NULL,
  PRIMARY KEY (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8;

CREATE TABLE `user_to_account` (
  `user_id` varchar(36) CHARACTER SET ascii NOT NULL,
  `account_id` varchar(36) CHARACTER SET ascii NOT NULL,
  KEY `user_to_account_ibfk_1_idx` (`user_id`),
  KEY `user_to_account_ibfk_2_idx` (`account_id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8;
```

```php
$resultSet = $this->getModelsManager()->createBuilder()
            ->columns('users.*')   // Errors unless "userAccount.*" or "userAccount.{random-column}"
            ->addFrom('Models\UserAccount', 'userAccount')
            ->leftJoin('Models\User','users.id = userAccount.userId', 'users')
            ->getQuery()
            ->execute();
```

Error:
**Column 'id' doesn't make part of the column map**

Let me know if I can provide more information that would be useful ...

**Comment by virgofx** [comment]
More information. It appears the error is thrown **after** the execute and during the enumeration (foreach) of the result set:

```php
        foreach ($resultSet as $object) {  // error thrown here
            // Attempt to see if the $object is a User or Robot
            // Not getting here because of error when translating the $resultSet
            var_dump ($object);
        }
```

**Comment by dreamsxin** [comment]
Use `$row->r->id`

**Comment by virgofx** [comment]
@dreamsxin Can you clarify with respect to above example? The error would be thrown prior to retrieving the row...

**Comment by dreamsxin** [comment]
@virgofx It's I test this code result
```php
$resultSet = $modelsManager->createBuilder()
        ->columns('r.*')
        ->addFrom('Some\RobotsParts', 'rp')
        ->leftJoin('Some\Robots','r.id = rp.robots_id', 'r')
        ->getQuery()
        ->execute();
```
You code should be right, Are you pull the lasted zephir to install?

**Comment by rkeplin** [comment]
I am able to reproduce this.  When the following is executed, a Phalcon\Mvc\Model\Exception is thrown with the message "Column 'streetAddress' doesn't make part of the column map"

```php
$row = $this->modelsManager->createBuilder()
    ->columns('Address.*')
    ->from('Office')
    ->join('Address', 'Office.addressId = Address.id')
    ->where('Office.id = 1')
    ->getQuery()
    ->getSingleResult();
```

The following does not throw any exceptions, and returns an Office:
```php
$row = $this->modelsManager->createBuilder()
            ->columns('Office.*')
            ->from('Office')
            ->join('Address', 'Office.addressId = Address.id')
            ->where('Office.id = 1')
            ->getQuery()
            ->getSingleResult();
```

And then the following returns false, when it should return something.
```php
$row = $this->modelsManager->createBuilder()
            ->columns('Office.*, Address.id')
            ->from('Office')
            ->join('Address', 'Office.addressId = Address.id')
            ->where('Office.id = 1')
            ->getQuery()
            ->getSingleResult();
```

**Comment by KorsaR-ZN** [comment]
@rkeplin This has been fixed, try the latest version of the 2.0.x branch (see #10439)

**Comment by rkeplin** [comment]
Thanks, @KorsaR-ZN

**Comment by virgofx** [comment]
Confirm fixed in 2.0.3. :+1:
